internal/backend/sessions/sessionworkflows/workflows.go

```diff
@@ -195,32 +193,42 @@ func (ws *workflows) sessionWorkflow(wctx workflow.Context, params *sessionWorkf
 	//                blow us up due to non determinism.
 
 	var prints []string
+	var duration time.Duration
+	var startTime time.Time
+	fields := []zap.Field{}
 
 	data, err := ws.getSessionData(wctx, params.SessionID)
 	if err == nil {
-		if workflow.IsReplaying(wctx) && data.Session.State().IsFinal() {
-			// HACK: If we somehow get a signal to this workflow after it completed,
-			//       in certain delicate timing, the workflow rekicks in replay.
-			//       Though replay would work just fine, and result in a no-op,
-			//       it's nice to not have to run through its entirely. Temporal
-			//       just seems to ignore it...
-			z.Info("already completed")
-			return nil, nil
-		}
 
-		sessionsCreatedCounter.Add(ctx, 1)
+		if workflow.IsReplaying(wctx) {
+			if data.Session.State().IsFinal() {
+				// HACK: If we somehow get a signal to this workflow after it completed,
+				//       in certain delicate timing, the workflow rekicks in replay.
+				//       Though replay would work just fine, and result in a no-op,
+				//       it's nice to not have to run through its entirely. Temporal
+				//       just seems to ignore it...
+				l.Info("already completed")
+				return nil, nil
+			}
+			fields = append(fields, zap.Int32("attempt", wi.Attempt))
+		} else {
+			sessionsCreatedCounter.Add(ctx, 1)
+		}
 
-		z.Info("session workflow: started")
-		prints, err = runWorkflow(wctx, z, ws, data, params.Debug)
+		startTime = data.Session.CreatedAt()
+		l.Info("session workflow: started", fields...)
+		prints, err = runWorkflow(wctx, l, ws, data, params.Debug)
```

Comment:
I'm not sure, 
do you know that if we pass this line, the session is for sure done?
if error is returned there would be no retry or something?
and what about if there server crashed before updating the session state to a final state in db?

i prefer to lost a metric than having an incorrect metric

could you verify this please?

Reply:
talked